client/state/selectors/has-site-comments.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * External dependencies
+ */
+import { get, find } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import createSelector from 'lib/create-selector';
+
+/**
+ * Returns true if we have site comments
+ *
+ * @param  {Object}  state       Global state tree
+ * @param  {Number}  siteId      The ID of the site we're querying
+ */
+export const hasSiteComments = createSelector(
+	( state, siteId ) => {
+		const comments = get( state, 'comments.items', {} );
+		return !! find( Object.keys( comments ), key => parseInt( key.split( '-', 1 ), 10 ) === siteId );
+	},
+	state => [ state.comments.items ]
```

Comment:
Oh I misread the code. The shallow equals is for the list of dependents rather for a single state object. So we should be OK on reference equality.

> this code will iterate over the new and old items object on every call,

I'm not sure what you mean for this scenario.